Remove search autosuggests

nikomakela opened the pull request:
HH-372.

Remove search autosuggest component from all apps and replace it with a similar looking text input, without the auto suggest menu.

Other miscallenous improvements:
1. Rename the state and reference handlers related to the search input.
2. Make code more reusable by moving it to packages/component
3. Provide a generic advanced search (form) input component (wrapper), that handles styling and adding of icon for a search input.

nikomakela committed:
feat(search): replace autosuggest field with simpler text search field

HH-374.

Remove the autosuggest (menu) components, styles and types. Replace the
autosuggestion search fields (on landing page and search page) with a
new shared text search field, which is simpler, does not have any
autocomplete feature, but still looks the same as the field used
earlier.

nikomakela committed:
refactor(search): use generic element to create advanced search inputs

HH-374.

Add AdvancedSearchInput to deal with multiple different kind of
components and to provide container and icon elements for the input
component. AdvancedSearchInput is a component that wraps an icon and an
input field.

nikomakela committed:
test(search): advanced search text input

HH-374

nikomakela committed:
test(search): advanced search input

HH-374

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr839.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __failed__ for https://harrastukset-pr839.dev.hel.ninja :crying_cat_face::anger::boom::boom:

nikomakela committed:
fix(search): add form element to landing page search

HH-374.

Add a form element around the landing page search form component (as a
container), so the submitting behaviours of the form would act as
usually. E.g. the enter pressing should usually submit the form.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr839.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr839.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr839.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr839.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr839.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr839.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders reviewed (approved):
Placeholder text should be fixed in liikunta. Left some comments, please take a look at them.

nikomakela committed:
style(sonar): fix unnecessary spread object literal

HH-374

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr839.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr839.dev.hel.ninja :laughing::tada::tada::tada:
